src/metabase/automagic_dashboards/core.clj

```diff
@@ -779,23 +779,15 @@
           [(collect-metrics root question)
            (collect-breakout-fields root question)])))
 
-(defn- key-in?
-  "Recursively finds key in coll, returns true or false"
-  [coll k]
-  (boolean (let [coll-zip (zip/zipper coll? #(if (map? %) (vals %) %) nil coll)]
-             (loop [x coll-zip]
-               (when-not (zip/end? x)
-                 (if (k (zip/node x)) true (recur (zip/next x))))))))
-
 (defn- splice-in
-  [join-statement card-member]
-  (let [query (get-in card-member [:card :dataset_query :query])]
-    (if (key-in? query :join-alias)
-      ;; Always in the top level even if the join-alias is found deep in there
-      (assoc-in card-member [:card :dataset_query :query :joins] join-statement)
-      card-member)))
-
-(defn- maybe-enrich-joins
+  "If the dashcard is a query card, ensure the joins are present."
+  [join-statement dashcard]
```

Comment:
`splice-in`'s args feel like they are in the wrong order to me. it is kind of an assoc, so I would have expected the thing you are assoc-ing onto to come first.

Reply:
Once you collapse the splice fn down to one form it makes sense to just collapse it all down into `preserve-joins`.